chain/stmgr/stmgr.go

```diff
@@ -136,6 +139,10 @@ type StateManager struct {
 	tsExec        Executor
 	tsExecMonitor ExecMonitor
 	beacon        beacon.Schedule
+
+	// We keep a small cache for calls to ExecutionTrace which helps improve
+	// performance for node operators like exchanges and block explorers
+	execTraceCache *lru.ARCCache[*types.TipSet, tipSetCacheEntry]
```

Comment:
A pointer type like this won't work for a key (this will compare pointers, which will most certainly be different at insertion and access times, even if they point to values that are deeply equal). Let's use the TipsetCid instead.